### ISSUE TITLE
OutOfMemoryError: Failed to allocate a 67644144 byte allocation with 25165824 free bytes and 61MB until OOM, max al...

### ISSUE THREAD

**Issue by pocmo** [mozilla-mobile/android-components#5744]
https://sentry.prod.mozaws.net/operations/fenix-fennec-nightly/issues/7170540/

```
OutOfMemoryError: Failed to allocate a 67644144 byte allocation with 25165824 free bytes and 61MB until OOM, max allowed footprint 229046520, growth limit 268435456
    at java.lang.StringFactory.newStringFromChars(StringFactory.java:267)
    at java.lang.StringBuffer.toString(StringBuffer.java:671)
    at java.io.StringWriter.toString(StringWriter.java:210)
    at kotlin.io.FilesKt__FileReadWriteKt.readText(FileReadWrite.kt:5)
    at kotlin.io.FilesKt__FileReadWriteKt.readText$default(FileReadWrite.kt:2)
...
(8 additional frame(s) were not displayed)

Failed to allocate a 67644144 byte allocation with 25165824 free bytes and 61MB until OOM, max allowed footprint 229046520, growth limit 268435456
```

**Comment by pocmo** [comment]
`OutOfMemoryError` during open tabs migration. This user must have a lot of open tabs. It is only happening for one user but a lot of times.. so I assume we are unable to successfully migrate and keep retrying on the next start.

**Comment by abodea** [comment]
Verified as fixed using **Fennec Nightly Migration Build 7** using the following devices:
- Google Pixel XL (Android 10) - ARM64
- LG G4 (Android 6.0.1) - ARM 
- Xiaomi Mi Pad 2 (Android 5.1) - x86 architecture
- Samsung Galaxy S10+(Android 9) - ARM64
- Google Pixel 3 XL(Android 9) - ARM64

Note that I used more than 50 tabs with websites having large images.
Everything was correctly migrated and the website numbers were correct.

@pocmo we will keep an eye on this issue for the future tests and get back in case we can find proper repro-steps or if we will be able to reproduce it.
